src/software/thunderscope/chicker/chicker.py

```diff
@@ -0,0 +1,227 @@
+                        button.toggle()
+                        self.charged = True
+                        # ----- replace this line ----- (next PR)
+                        print("Charge clicked")
+                        print("Geneva:", self.geneva_value, "Power:", self.power_value)
+
+                elif self.charged:
+                    # discharge button white; text: "Discharge"; setCheckable(True)
+                    if button.text() == "Discharge":
+                        button.setStyleSheet("background-color: White")
+                        button.setCheckable(True)
+                    # kick/chip button white; setCheckable(True)
+                    else:
+                        button.setStyleSheet("background-color: White")
+                        button.setCheckable(True)
+
+                    if button.isChecked():
+                        # discharge clicked; change text to 'Charge'; print; set 'No Auto' on; charged = False
+                        if button.text() == "Discharge":
+                            button.toggle()
+                            self.charged = False
+                            # ----- replace this line ----- (next PR)
+                            print("Discharge clicked")
+                            print(
+                                "Geneva:", self.geneva_value, "Power:", self.power_value
+                            )
+                        # kick/chip button clicked; change text to 'Charge';
+                        else:
+                            button.toggle()
+                            self.charged = False
+                            if button.text() == "Kick":
+                                # ----- replace this line ----- (next PR)
+                                print("Kick clicked")
+                                print(
+                                    "Geneva:",
+                                    self.geneva_value,
+                                    "Power:",
+                                    self.power_value,
+                                )
+                            elif button.text() == "Chip":
+                                # ----- replace this line ----- (next PR)
+                                print("Chip clicked")
+                                print(
+                                    "Geneva:",
+                                    self.geneva_value,
+                                    "Power:",
+                                    self.power_value,
+                                )
+
+            # check all radio buttons, regardless of self.charged value
+            for radio in item.findChildren(QRadioButton):
+                if radio.isChecked():
+                    # no auto clicked - change to no auto; Kick/chip white and enabled
+                    if radio.text() == "No Auto":
+                        self.radioCheckable[1] = True
+                        self.radioCheckable[2] = True
```

Comment:
we can actually add the radio buttons to a button group and we don't need to manually manage this state

checkout this example: https://stackoverflow.com/questions/14798058/grouping-radio-buttons-in-pyqt

Reply:
I added the button group, which is useful for managing the stats of all buttons. But to print the state change _only once_ I still need to use these, for some reason